wrangler-service/src/main/java/co/cask/wrangler/service/kafka/KafkaService.java

```diff
@@ -258,19 +284,89 @@ public void read(HttpServiceRequest request, HttpServiceResponder responder,
         object.addProperty(PropertyIds.ID, uuid);
         object.addProperty(PropertyIds.NAME, topic);
         values.add(object);
-        response.addProperty("status", HttpURLConnection.HTTP_OK);
-        response.addProperty("message", "Success");
-        response.addProperty("count", values.size());
-        response.add("values", values);
+        response.addProperty(PropertyIds.STATUS, HttpURLConnection.HTTP_OK);
+        response.addProperty(PropertyIds.MESSAGE, PropertyIds.SUCCESS);
+        response.addProperty(PropertyIds.COUNT, values.size());
+        response.add(PropertyIds.VALUES, values);
         sendJson(responder, HttpURLConnection.HTTP_OK, response.toString());
-      } finally {
-        consumer.close();
-      }
-    } catch (Exception e) {
+    } catch (Throwable e) {
+      LOG.error("Exception while reading data from kafka topic", e);
       error(responder, e.getMessage());
     }
   }
+  
+  
+  
+  public void generateSchema(KafkaConfiguration config, String format) throws Exception {
+	Properties properties = config.get();
+	if (!format.equalsIgnoreCase(Formats.AVRO) && !format.equalsIgnoreCase("binary")) {
+		outputFields = new ArrayList<>();
+		outputFields.add("body");
+		outputSchema = Format.TEXT.getSchema().toString();
+		return;
+	} else if ((format.equalsIgnoreCase(Formats.AVRO) && !properties.containsKey(SchemaRegistryClient.Configuration.SCHEMA_REGISTRY_URL.name())) || format.equalsIgnoreCase("binary")){
+		outputFields = new ArrayList<>();
+		outputFields.add("body");
+		outputSchema = Format.BLOB.getSchema().toString();
+		return;
+	}
+    String schemaName = properties.getProperty(PropertyIds.SCHEMA_NAME);
+    if (schemaName == null || schemaName.trim().isEmpty()) {
+      throw new IllegalArgumentException("Empty schema name");
+    }
+
+    String schemaRegistryUrl = properties.getProperty(SchemaRegistryClient.Configuration.SCHEMA_REGISTRY_URL.name());
+    LOG.error("Schema Registry Url: {}", schemaRegistryUrl);
+    if (schemaRegistryUrl == null || schemaRegistryUrl.trim().isEmpty()) {
+      throw new IllegalArgumentException("Empty schema registry url");
+    }
+
+    schemaRegistryUrl = schemaRegistryUrl.trim();
+    if (!schemaRegistryUrl.endsWith("/")) {
+      schemaRegistryUrl = schemaRegistryUrl + "/";
+    }
+
+    schemaRegistryUrl = schemaRegistryUrl + "schemaregistry/schemas/" + schemaName + "/versions/latest";
+
+    String schemaString = "";
+    String keytabLocation = config.getKeytabLocation();
+    String principal = config.getPrincipal();
+    if(Strings.isNullOrEmpty(keytabLocation) || Strings.isNullOrEmpty(principal)) {
+       schemaString = URLExtractor.extract(schemaRegistryUrl, 30);
+    } else {
+       schemaString = KerberosHttpClient.sendKerberisedGetRequest(schemaRegistryUrl,
+    		   principal, keytabLocation);
+    }
+    parseSchemaString(schemaString);
+  }
 
+  private void parseSchemaString(String schemaString) {
```

Comment:
Its always better to return value instead of functions returning void. We can return outputSchema here.